flink-runtime/src/test/java/org/apache/flink/runtime/jobmaster/JobManagerRunnerTest.java

```diff
@@ -204,11 +211,52 @@ public void testLibraryCacheManagerRegistration() throws Exception {
 		}
 	}
 
+	/**
+	 * Tests that the {@link JobManagerRunner} always waits for the previous leadership operation
+	 * (granting or revoking leadership) to finish before starting a new leadership operation.
+	 */
+	@Test
+	public void testConcurrentLeadershipOperations() throws Exception {
+		final CompletableFuture<Acknowledge> suspendedFuture = new CompletableFuture<>();
+
+		TestingJobMasterServiceFactory jobMasterServiceFactory = new TestingJobMasterServiceFactory(
+			() -> new TestingJobMasterService(
+				"localhost",
+				e -> suspendedFuture));
+		JobManagerRunner jobManagerRunner = createJobManagerRunner(jobMasterServiceFactory);
+
+		jobManagerRunner.start();
+
+		leaderElectionService.isLeader(UUID.randomUUID()).get();
+
+		leaderElectionService.notLeader();
```

Comment:
do we also want to test `isLeader` against `jobMasterService.start` future in the similar way?

Reply:
Yes I think it would makes sense. Will add the test.